src/mcts/search.cc

```diff
@@ -648,7 +653,10 @@ std::vector<EdgeAndNode> Search::GetBestChildrenNoTemperature(Node* parent,
           // This default isn't used as wl only checked for case edge is
           // terminal.
           const auto wl = edge.GetWL(0.0f);
-          if (!edge.IsTerminal() || !wl) return kNonTerminal;
+          // Not safe to access IsTerminal if GetN is 0.
+          if (edge.GetN() == 0 || !edge.IsTerminal() || !wl) {
```

Comment:
Maybe the check should be added to `EdgeAndNode::IsTerminal()`?

Reply:
I 100% am onboard with that, but I'm not on board with having to validate all the existing uses of EdgeAndNode::IsTerminal to see which ones fall into code paths that are allowed to check IsTerminal... So I went with the 'safer' change.